src/components/GlobalLearningResourcesPage/GlobalLearningResourcesFiltersCategory.tsx

```diff
@@ -9,18 +9,45 @@ import {
   TextVariants,
 } from '@patternfly/react-core';
 import { FiltersCategory } from '../../utils/FiltersCategoryInterface';
+import { Filter } from '../../utils/filtersInterface';
+import { loaderOptionsDefault } from '../../utils/fetchAllData';
 
 const GlobalLearningResourcesFiltersCategory: React.FC<FiltersCategory> = ({
   categoryId,
   categoryName,
   categoryData,
+  loaderOptions,
+  setLoaderOptions,
 }) => {
   const [isExpanded, setIsExpanded] = useState(true);
 
   const onToggle = (_event: React.MouseEvent, isExpanded: boolean) => {
     setIsExpanded(isExpanded);
   };
 
+  const updateLoaderOptions = (filter: Filter, isChecked: boolean) => {
+    const updatedCategory = isChecked
+      ? [
+          ...(loaderOptions[categoryId as keyof typeof loaderOptionsDefault] ||
```

Comment:
Rather than this typecasting, can't we define the `categoryId` as the type? All the way to the API request?